lib/puppet_x/puppetlabs/aws.rb

```diff
@@ -31,8 +35,16 @@ def self.read_only(*methods)
         end
       end
 
+      def self.logger
```

Comment:
Could we pull the addition of logging out to a separate commit, since it doesn't really belong with the memoization?